test/unit/compare-tree/compare-tree.test.js

```diff
@@ -0,0 +1,330 @@
+import CompareTree from '../../../src/compare-tree'
+import { set } from '../../../src/utils'
+import SOLAR_SYSTEM from './fixtures/solar-system'
+
+describe('CompareTree', function () {
+  beforeEach(function () {
+    this.tree = new CompareTree(SOLAR_SYSTEM)
+  })
+
+  describe('::on', function () {
+    it('passes the selected state into a compare', function () {
+      expect.assertions(1)
+
+      let next = set(SOLAR_SYSTEM, ['meta', 'selected'], 'jupiter')
+      let handler = jest.fn()
+
+      this.tree.on('meta.selected', handler)
+      this.tree.update(next)
+
+      expect(handler).toHaveBeenCalledWith('jupiter')
+    })
+
+    it('passes every key in the query', function () {
+      expect.assertions(1)
+
+      let next = set(SOLAR_SYSTEM, ['meta', 'selected'], 'jupiter')
+      let handler = jest.fn()
+
+      this.tree.on('meta.selected,planets', handler)
+      this.tree.update(next)
+
+      expect(handler).toHaveBeenCalledWith('jupiter', SOLAR_SYSTEM.planets)
+    })
+
+    it('returns the same node given the same path', function () {
+      expect.assertions(1)
+
+      let a = this.tree.on('meta.selected', jest.fn())
+      let b = this.tree.on('meta.selected', jest.fn())
+
+      expect(a).toBe(b)
+    })
+
+    it('returns the same node given the same path in a different order', function () {
+      expect.assertions(1)
+
+      let a = this.tree.on('planets,meta.selected', jest.fn())
+      let b = this.tree.on('meta.selected,planets', jest.fn())
+
+      expect(a).toBe(b)
+    })
+
+    it('only invokes a CompareTree once if given multiple compares', function () {
+      let a = set(SOLAR_SYSTEM, ['meta', 'selected'], 'jupiter')
+      let b = set(a, ['meta', 'focused'], false)
+
+      let handler = jest.fn()
+
+      this.tree.on(['meta.selected', 'meta.focused'], handler)
+      this.tree.update(b)
+
+      expect(handler).toHaveBeenCalledTimes(1)
```

Comment:
Test name might be from a bad find/replace. This is testing to make sure that, when a subscription listens to two key paths, and both of them change at once, it only fires the callback a single time.